src/snowflake/snowpark/modin/pandas/base.py

```diff
@@ -442,19 +442,32 @@ def _binary_op(
         # pandas itself will ignore the axis argument when using Series.<op>.
         # Per default, it is set to axis=0. However, for the case of a Series interacting with
         # a DataFrame the behavior is axis=1. Manually check here for this case and adjust the axis.
-        if isinstance(self, pd.Series) and isinstance(other, pd.DataFrame):
-            axis = 1
+
+        is_lhs_series_and_rhs_dataframe = (
```

Comment:
I'm not clear what happens when the order is reversed: lhs is dataframe and lhs is series. I can see that those cases are being tested (both when the op is commutative and not commutative), but I don't see how it's handled in the code.